api/envoy/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto

```diff
@@ -298,7 +298,7 @@ message HttpConnectionManager {
 
   // An optional override that the connection manager will write to the server
   // header in responses. If not set, the default is *envoy*.
-  string server_name = 10;
+  string server_name = 10 [(validate.rules).string = {pattern: "^[^\000\r\n]*$"}];
```

Comment:
nit: there is a pattern for this already `HTTP_HEADER_VALUE`

Reply:
I agree but the configuration should also have such constraints. Currently, server_name allow \0, \n, and \r which should not be there.

Reply:
I think @asraa is suggesting to use this pattern in the validator, see https://github.com/envoyproxy/envoy/blob/b6c8bb3a4ac6bcce221643a4924befd5eefd6815/api/envoy/config/core/v3/health_check.proto#L90 for an example.

Reply:
@htuch @asraa Thanks for the review. PTAL.